research/conditional/utils/argparse.py

```diff
@@ -4,74 +4,116 @@
-    parser.add_argument("--logging_interval_light", type=int, default=1000000)
-    parser.add_argument("--logging_interval_heavy", type=int, default=1000000)
+    parser.add_argument("--loss_checkpoint_chungs", type=int, default=0)
+    parser.add_argument("--data_distributed", action="store_true")
+    parser.add_argument(
+        "--model_parallelism_fragmentation",
+        type=str,
+        default=None,
+        help="comma-separated list of integers, that signify the numbers of model blocks that are first on the new device, e.g. 2,4 means that blocks 0,1 will be on GPU 0, blocks 2,3 will be on GPU 1, and the rest will be on GPU 2",
+    )
+    parser.add_argument("--detect_anomaly", action="store_true")
+
+    # CORE data hyperparameters, almost always specified in baseline configs
+
+    parser.add_argument(
+        "--dataset_type", type=str, choices=["wikibook", "c4"], required=True
+    )
+    parser.add_argument("--batch_size", type=int, required=True)
+    parser.add_argument("--cutoff", type=int, required=True)
+
+    # other data hyperparameters
+    parser.add_argument("--num_workers", type=int, default=8)
+
+    # other parameters usually not changed for experiments
+
     parser.add_argument("--mask_loss_weight", type=float, default=1.0)
     parser.add_argument("--mask_percent", type=float, default=0.15)
-    parser.add_argument("--n_steps", type=int, default=90000)
     parser.add_argument("--data_seed", type=int, default=42)
     parser.add_argument("--torch_seed", type=int, default=42)
-    parser.add_argument("--deterministic_experiment", action="store_true")
-    parser.add_argument("--tags", nargs="*", type=str, default=None)
-    parser.add_argument("--project_name", type=str, default="pmtest/llm-random")
-    parser.add_argument(
-        "--model_type", type=str, choices=["gpt", "bert"], default="bert"
-    )
-    # parameters usually changed for experiments
-    parser.add_argument("--ff_mode", type=str, default="vanilla")
-    parser.add_argument("--name", type=str, default="")
-    parser.add_argument("--learning_rate", type=float, default=3e-4)
-    parser.add_argument("--gradient_checkpointing", action="store_true")
+
+    # model versioning
+
     parser.add_argument("--save_weights_path", type=str, default=None)
     parser.add_argument("--save_weights_interval", type=int, default=1000)
     parser.add_argument("--load_weights_path", type=str, default=None)
-    parser.add_argument("--grad_clip", type=float, default=None)
-    parser.add_argument("--weight_decay", type=float, default=0.0)
-    parser.add_argument("--adam_beta1", type=float, default=0.9)
-    parser.add_argument("--adam_beta2", type=float, default=0.999)
-    parser.add_argument("--no_ff", action="store_true")
-    parser.add_argument("--loss_checkpoint_chungs", type=int, default=0)
-    parser.add_argument("--gradient_accumulation_steps", type=int, default=1)
-    parser.add_argument("--auto_find_grad_accumulation", action="store_true")
```

Comment:
this parameter got removed completely?

Reply:
correct, though it shouldve been done when I removed the logic for that as well. An oversight on my side.